src/shared/setup.js

```diff
@@ -46,6 +47,43 @@ async function connectDatabase () {
   return Db.connect(Config.DATABASE)
 }
 
+/**
+ * @function createSettlementModels
+ *
+ * @description Creates settlement models defined by config is not exists in database
+ *
+ * @param {object[]} settlementModels List of settlement models
+ */
+
+const createSettlementModels = async function (settlementModels = []) {
+  try {
+    // Get the existing settlement models
+    const url = `${Config.CENTRAL_LEDGER_ENDPOINT}/settlementModels`
+    const opts = { url, method: 'GET' }
+    const response = await request(opts)
+    const existingSettlementModels = response.data
+
+    if ((settlementModels == null || settlementModels.length < 1) && existingSettlementModels.length < 1) {
+      const error = 'Need to configure at least one settlement model'
+      Logger.error(error)
+      throw ErrorHandling.Factory.createFSPIOPError(ErrorHandling.Enums.FSPIOPErrorCodes.INTERNAL_SERVER_ERROR, error)
+    }
+
+    // Find any new settlement models to be added
+    const newSettlementModels = settlementModels.filter(a => !existingSettlementModels.some(b => b.name === a.name))
```

Comment:
refactor into small function (naming should help clarify) without need for comments